system/utils.py

```diff
@@ -205,28 +262,99 @@ def wrapped(operations, queue_size, add_size, get_size, event_loop):
     add = await write_eventually_positive(send_nym, pool_handle, wallet_handle, trustee_did, some_did)
     assert add['op'] == 'REPLY'
     get = await read_eventually_positive(get_nym, pool_handle, wallet_handle, trustee_did, some_did)
     assert get['result']['seqNo'] is not None
 
 
-async def check_ledger_sync():
-    hosts = [testinfra.get_host('ssh://node{}'.format(i)) for i in range(1, 8)]
+# TODO make that async
+def check_no_failures(hosts):
+    for host in hosts:
+        result = host.run('journalctl -u indy-node.service -b -p info')
+        assert result.find("indy-node.service: Failed") == -1, (
+            "Node service on host{} failed:\n{}".format(host.id, result)
+        )
+
+# TODO make that async
+async def check_ledger_sync(node_ids=None, nodes_num=7):
+    if node_ids is None:
+        node_ids = [(i + 1) for i in range(nodes_num)]
+    hosts = [NodeHost(i) for i in node_ids]
+
     pool_results = [host.run('read_ledger --type=pool --count') for host in hosts]
-    print('\nPOOL LEDGER SYNC: {}'.format([result.stdout for result in pool_results]))
+    print('\nPOOL LEDGER SYNC: {}'.format([result for result in pool_results]))
     config_results = [host.run('read_ledger --type=config --count') for host in hosts]
-    print('\nCONFIG LEDGER SYNC: {}'.format([result.stdout for result in config_results]))
+    print('\nCONFIG LEDGER SYNC: {}'.format([result for result in config_results]))
     domain_results = [host.run('read_ledger --type=domain --count') for host in hosts]
-    print('\nDOMAIN LEDGER SYNC: {}'.format([result.stdout for result in domain_results]))
+    print('\nDOMAIN LEDGER SYNC: {}'.format([result for result in domain_results]))
     audit_results = [host.run('read_ledger --type=audit --count') for host in hosts]
-    print('\nAUDIT LEDGER SYNC: {}'.format([result.stdout for result in audit_results]))
+    print('\nAUDIT LEDGER SYNC: {}'.format([result for result in audit_results]))
+
+    assert all([pool_results[i] == pool_results[i + 1] for i in range(-1, len(pool_results) - 1)])
+    assert all([config_results[i] == config_results[i + 1] for i in range(-1, len(config_results) - 1)])
+    assert all([domain_results[i] == domain_results[i + 1] for i in range(-1, len(domain_results) - 1)])
+    assert all([audit_results[i] == audit_results[i + 1] for i in range(-1, len(audit_results) - 1)])
+
+
+async def check_pool_performs_write_read(pool_handle, wallet_handle, trustee_did, nyms_count=1):
+    for i in range(nyms_count):
+        some_did, _ = await did.create_and_store_my_did(wallet_handle, '{}')
+        add = await send_nym(pool_handle, wallet_handle, trustee_did, some_did)
+        assert add['op'] == 'REPLY'
+        get = await get_nym(pool_handle, wallet_handle, trustee_did, some_did)
+        assert get['result']['seqNo'] is not None
+
+
+async def check_pool_is_workable(pool_handle, wallet_handle, trustee_did, nyms_count=3):
```

Comment:
I'd prefer `check_pool_is_functional`, we have similar helper function in plenum with similar name.